haystack/preview/document_stores/memory/document_store.py

```diff
@@ -47,6 +47,13 @@ def __init__(
         self.bm25_algorithm = algorithm_class
         self.bm25_parameters = bm25_parameters or {}
 
+        # Used to convert this instance to a dictionary for serialization
+        self.init_parameters = {
+            "bm25_tokenization_regex": bm25_tokenization_regex,
+            "bm25_algorithm": bm25_algorithm,
+            "bm25_parameters": self.bm25_parameters,
+        }
```

Comment:
Or we could use the same decorator that Canals components use? I understand you wanted to change this mechanic, so if it's just temporary no problem for this

Reply:
Nope, I plan to remove that is way too hidden to the user and prone to error.

It creates in the `init_parameters` dictionary given the keyword arguments provided when instantiating the component, that means that the below code would generate two different dictionaries. That's bad.

```
MemoryDocumentStore(bm25_tokenization_regex="some-regex" bm25_algorithm="BM25Plus", bm25_parameters={"key": "value"})

MemoryDocumentStore("some-regex", "BM25Plus", {"key": "value"})
```

Reply:
Alright! Sounds good then.